### PR TITLE
REGR: Fix nan comparison for same Index object

### LINKED ISSUE #47105: BUG: nan comparisons behave differently on identical objects
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
>>> import pandas as pd
>>> import numpy as np
>>> idx = pd.Index([np.nan])
>>> idx > idx
array([ True])
>>> idx > idx.copy()
array([False])
```

### Issue Description

When an index contains `np.nan`, it compares differently to itself than to any other index object containing `np.nan` (including a copy of itself). This seems likely to be caused by some internal check for identity (`other is self`) that is taking precedence over element-wise comparisons.

### Expected Behavior

NaN should follow IEEE 754, so we should see
```
>>> idx > idx
array([False])
```
without needing a copy.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 4bfe3d07b4858144c219b9346329027024102ab6
python           : 3.8.13.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-76-generic
Version          : #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_US.UTF-8
pandas           : 1.4.2
numpy            : 1.21.6
pytz             : 2022.1
dateutil         : 2.8.2
pip              : 22.1.1
setuptools       : 62.3.2
Cython           : 0.29.30
pytest           : 7.1.2
hypothesis       : 6.46.7
sphinx           : 4.5.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.8.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 8.3.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           :
fastparquet      : None
fsspec           : 2022.5.0
gcsfs            : None
markupsafe       : 2.1.1
matplotlib       : None
numba            : 0.55.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 7.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.5.0
scipy            : 1.8.1
snappy           :
sqlalchemy       : 1.4.36
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #47105 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -6955,7 +6955,7 @@ def _cmp_method(self, other, op):
                     # TODO: should set MultiIndex._can_hold_na = False?
                     arr[self.isna()] = False
                 return arr
-            elif op in {operator.ne, operator.lt, operator.gt}:
+            elif op == operator.ne:
```
Comment (simonjayhawkins) [inline pandas/core/indexes/base.py:6958]: is?

Comment (phofl) [reply pandas/core/indexes/base.py:6958]: Was not sure about it, but it does get hit through !=

Comment (simonjayhawkins) [reply pandas/core/indexes/base.py:6958]: quickly scanning the code, it appears we mostly use `is` elsewhere

Comment (phofl) [reply pandas/core/indexes/base.py]: done

File: pandas/tests/indexes/test_base.py [hunk 2]
```diff
@@ -1604,3 +1604,15 @@ def test_get_attributes_dict_deprecated():
     with tm.assert_produces_warning(DeprecationWarning):
         attrs = idx._get_attributes_dict()
     assert attrs == {"name": None}
+
+
+def test_nan_comparison_same_object():
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py:1609]: can you test gt as well as lt; this doesn't affect le, ge right?

Comment (phofl) [reply pandas/tests/indexes/test_base.py]: Done and no, they are checked i a different if block

**Review by simonjayhawkins — APPROVED** [review]
thanks @phofl lgtm.

**Comment by jreback** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**